### PR TITLE
Update the docs based on telecon discussion

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

- Split out the known telescopes and utility functions as their own sections in the docs (not in developer docs)
- Organize and provide more context in developer docs
- change all docs organization to be in code order rather than in alphabetical order
- include all classes (except UVFlag) -- some were missing before
- fix a couple of readme details we discussed on the last telecon
- update a few docstrings

see RTD build at: https://pyuvdata.readthedocs.io/en/docs_update/index.html

## Motivation and Context

Some classes and functions were not included in our docs, developer docs weren't well organized and included some things (like utils) that we thought should be elevated a little.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Documentation change checklist:
- [x] Any updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] If this is a significant change to the readme or other docs, I have checked that they are rendered properly on ReadTheDocs. (you may need help to get this branch to build on RTD, just ask!)

### CODE REVIEW

File: docs/developer_docs.rst [hunk 1]
```diff
@@ -30,24 +48,34 @@ Classes
...
+
+.. autofunction:: pyuvdata.fhd.get_fhd_history
 
+aipy extracts
+_____________
 
-Utility Functions
------------------
-.. automodule:: utils
+.. automodule:: pyuvdata.aipy_extracts
```
Comment (adampbeardsley) [inline docs/developer_docs.rst]: I don't think these are rendering correctly on RTD:
image

Comment (bhazelton) [reply docs/developer_docs.rst]: oops, good catch! It should be fixed now (as soon as RTD updates)

File: pyuvdata/aipy_extracts.py [hunk 2]
```diff
@@ -521,11 +652,23 @@ def write(self, preamble, data, flags=None):
...
-        omitted completely.
-
+        """
+        Initialize header items and variables from another UV.
+
+        Those in override will be overwritten by override[k], and tracking will
+        be turned off (meaning they will not be updated in pipe()). Those in
+        exclude are omitted completely.
+
+        TODO: check that defaults work properly. I usually use None in this context
```
Comment (adampbeardsley) [inline pyuvdata/aipy_extracts.py]: Do we want to leave a TODO in the code? Can we make this an issue instead?
The sentence "I usually use None in this context" is also ambiguous and I'm not sure how the user is supposed to use that.

Comment (bhazelton) [reply pyuvdata/aipy_extracts.py]: you're right, this isn't very helpful. I just made an issue, I'll take the TODO out of the code now.

**Pull request merged**